civictechprojects/models.py

```diff
@@ -39,6 +39,39 @@ class Project(models.Model):
     project_url = models.CharField(max_length=200, blank=True)
     project_links = models.CharField(max_length=5000, blank=True)
 
+    def hydrate_to_json(self):
+        files = ProjectFile.objects.filter(file_project=self.id)
+        thumbnail_files = list(files.filter(file_category=FileCategory.THUMBNAIL.value))
+        other_files = list(files.filter(file_category=FileCategory.ETC.value))
+        links = ProjectLink.objects.filter(link_project=self.id)
+
+        project = {
+            'project_id': self.id,
+            'project_name': self.project_name,
+            'project_creator': self.project_creator.id,
+            'project_description': self.project_description,
+            'project_url': self.project_url,
+            'project_location': self.project_location,
+            'project_issue_area': Tag.hydrate_to_json(list(self.project_issue_area.all().values())),
+            'project_files': list(map(lambda file: file.to_json(), other_files)),
+            'project_links': list(map(lambda link: link.to_json(), links))
+        }
+
+        if len(thumbnail_files) > 0:
+            project['project_thumbnail'] = thumbnail_files[0].to_json()
+
+        return project
+
+    # Remove any tags that aren't part of the canonical list
+    @staticmethod
+    def remove_tags_not_in_list():
+        for project in Project.objects.all():
+            for issue in project.project_issue_area.all():
+                issue_tag = Tag.get_by_name(issue)
+                if not issue_tag:
+                    print('Removing invalid tag', issue, 'from project:', project.project_name)
+                    project.project_issue_area.remove(issue)
```

Comment:
is it necessary to save this mutation?

Reply:
No.  Adding and removing tags doesn't seem to require saving the base Project.